plugins/woocommerce/src/Internal/ComingSoon/ComingSoonRequestHandler.php

```diff
@@ -26,6 +26,7 @@ class ComingSoonRequestHandler {
 	final public function init( ComingSoonHelper $coming_soon_helper ) {
 		$this->coming_soon_helper = $coming_soon_helper;
 		add_filter( 'template_include', array( $this, 'handle_template_include' ) );
+		add_filter( 'wp_theme_json_data_theme', array( $this, 'filter_theme_json_theme' ) );
```

Comment:
Feels awkward to have this in coming soon class since it'll be available for the entire site. Would it be ok temporarily? Or should we generalize for Woo core usage?

Reply:
Could we check that `woocommerce_coming_soon == 'yes'` or that the current page being viewed is coming soon or the template?

Reply:
@psealock I tried by checking the current page to `site-editor.php` and the current template, it worked to not load in non-coming soon page. However, due to the way site editor works, the loaded fonts persists even when we switch between templates 😞 

I don't think limiting the font load in site editor will be viable

Reply:
> Would it be ok temporarily? Or should we generalize for Woo core usage?

I think it's fine to have it in the coming soon class for now. We can always refactor it later if needed.

> however, due to the way site editor works, the loaded fonts persists even when we switch between templates, which means once we load coming soon page, the fonts will carry through other pages as well 

That makes sense. 😞 

As @psealock suggested, could we check that `woocommerce_coming_soon == 'yes'`? That way, we can ensure that the fonts are only loaded when the coming soon mode is enabled.

Reply:
> woocommerce_coming_soon == 'yes'?

@chihsuan @psealock Sorry I forgot to address that. I think it'll be a problem since we're checking this for both site-editor and frontend. While site live, users still can edit coming soon page, so the fonts will need to exist regardless of the coming soon state.

Reply:
I see. That makes sense!